backend/lib/services/members/MemberManager.js

```diff
@@ -187,35 +187,38 @@ class MemberManager {
     this.subjectList.delete(item.id)
   }
 
-  async deleteServiceAccountSecret (item) {
+  async deleteServiceAccountSecrets (item) {
     const { namespace } = Member.parseUsername(item.id)
     if (namespace !== this.namespace) {
       throw new UnprocessableEntity('It is not possible to modify a ServiceAccount from another namespace')
     }
 
-    const name = _
+    const names = _
       .chain(item)
       .get('extensions.secrets', [])
-      .tap(secrets => {
-        if (secrets.length > 1) {
-          throw new UnprocessableEntity(`ServiceAccount ${namespace} has more than one secret`)
-        }
-      })
-      .head()
-      .get('name')
+      .map('name')
       .value()
-    await this.client.core.secrets.delete(namespace, name)
+
+    _.forEach(names, async name => {
+      await this.client.core.secrets.delete(namespace, name)
```

Comment:
we did not have error handling but I think we should add some here - e.g. accept 404 errors as the user could add non existent secrets to the service account but return an error to the frontend in all other cases so that the user has an indication that the rotation did not succeed (completely). WDYT?

Reply:
All errors are forwarded to the outer async function here but you are right we not handle 404 like we do for delete serviceaccount.